src/routes/auth/admin.ts

```diff
@@ -19,28 +19,33 @@ adminAuthRouter.post("/login", async (req, res) => {
   if (!valid) return res.status(400).json(errors);
 
   // Authenticate user
-
-  //checking if the admin exists in DB
-  const adminLoginDoc = await AdminModel.findOne({ login: adminLoginData.login }).exec();
-  if (!adminLoginDoc) return res.status(400).json({ login: "email or password is incorrect" });
-  else if (!(await adminLoginDoc.checkPassword(adminLoginData.password))) {
-    return res.status(400).json({ login: "email or password is incorrect" });
+  try {
+    //checking if the admin exists in DB
+    const adminLoginDoc = await AdminModel.findOne({ login: adminLoginData.login }).exec();
+    if (!adminLoginDoc) return res.status(400).json({ login: "email or password is incorrect" });
+    else if (!(await adminLoginDoc.checkPassword(adminLoginData.password))) {
+      return res.status(400).json({ login: "email or password is incorrect" });
+    }
+
+    // Return auth token;
+    const accessToken = jwt.sign({ login: adminLoginDoc.login }, process.env.JWT_SECRET_ADMIN);
+
+    res.status(200).json({ jwt: `Bearer ${accessToken}` });
+  } catch (err) {
+    console.log(err);
   }
-
-  // Return auth token;
-  let accessToken;
-  const TOKEN = process.env.ACCESS_TOKEN_SECRET;
-  TOKEN ? (accessToken = jwt.sign({ login: adminLoginDoc.login }, TOKEN)) : console.log("No secret access token in env");
-
-  res.status(200).json({ jwt: `Bearer ${accessToken}` });
 });
 
 //admin whoami route
 adminAuthRouter.post("/me", async (req, res) => {
-  const adminDoc = await AdminModel.findOne({ login: res.locals.user.login }).exec(); // look for the user in db
-
-  // return data
-  res.status(200).json({
-    login: adminDoc?.login,
-  });
+  try {
+    const adminDoc = await AdminModel.findOne({ login: res.locals.user.login }); // look for the user in db
+
+    // return data
+    res.status(200).json({
+      login: adminDoc?.login,
+    });
+  } catch (err) {
+    console.log(err);
```

Comment:
Better to send an error to express (`next(error)`). You can use this to generate nice errors: https://www.npmjs.com/package/http-errors

Here it's also worse than before since you don't even res anything in case of error. Meaning if something goes wrong the request will never terminate.